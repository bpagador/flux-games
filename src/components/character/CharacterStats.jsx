import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';
import Stats from '../stats/Stats';



const CharacterStats = ({ character }) => (
  <section>
    <Character image={character.image} />
    <Stats
      food={character.food}
      water={character.water}
      movement={character.movement}
      gov={character.gov}
      total={character.total}
    />
  </section>
);

CharacterStats.propTypes = {
  character: PropTypes.shape({
    image: PropTypes.string.isRequired,
    food: PropTypes.number.isRequired,
    water: PropTypes.number.isRequired,
    movement: PropTypes.number.isRequired,
    gov: PropTypes.number.isRequired,
    total: PropTypes.number.isRequired
  })
};

export default CharacterStats;
