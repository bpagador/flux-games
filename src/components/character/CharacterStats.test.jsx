import React from 'react';
import { shallow } from 'enzyme';
import CharacterStats from './CharacterStats';


describe('CharacterStats component', () => {
  it('renders character with stats', () => {
    const character = {
      image: 'http://test.jpeg',
      food: 25,
      water: 25,
      movement: 25,
      gov: -50,
      total: 25
    };

    const wrapper = shallow(<CharacterStats character={character} />);
    expect(wrapper).toMatchSnapshot();
  });
});
