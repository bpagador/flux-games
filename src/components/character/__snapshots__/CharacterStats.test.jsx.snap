// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterStats component renders character with stats 1`] = `
<section>
  <Character
    image="http://test.jpeg"
  />
  <Stats
    food={25}
    gov={-50}
    movement={25}
    total={25}
    water={25}
  />
</section>
`;
